spec/index.bs

```diff
@@ -192,34 +192,81 @@ dictionary GPULimits {
 </script>
 
 
-Buffers {#buffers}
-==================
+# {{GPUBuffer}} # {#GPUBuffer}
 
-## GPUBuffer ## {#buffer}
+A {{GPUBuffer}} represents a linear allocation of memory to hold data that can be used by GPU operations.
+Each byte of the allocation can be addressed by its offset from the start of the {{GPUBuffer}}.
 
-<script type=idl>
-interface GPUBuffer : GPUObjectBase {
-    Promise<ArrayBuffer> mapReadAsync();
-    Promise<ArrayBuffer> mapWriteAsync();
-    void unmap();
+{{GPUBuffer}} has the following internal slots:
 
-    void destroy();
-};
-</script>
+<dl dfn-type=attribute dfn-for="GPUBuffer">
+	: <dfn>\[[size]]</dfn>
+	::
+        The Length of the {{GPUBuffer}} allocation in bytes, which is an unsigned long long.
+
+	: <dfn>\[[usage]]</dfn>
+	::
+        The allowed usages for this {{GPUBuffer}}, represented as a bitset unsigned long.
+
+    : <dfn>\[[state]]</dfn>
+    ::
+        This {{GPUBuffer}}'s state which is one of `"unmapped"`, `"mapped"`, `"destroyed"`.
+</dl>
+
+## Creation ## {#buffer-creation}
 
-### Creation ### {#buffer-creation}
+A {{GPUBuffer}} can be created unmapped using the {{GPUDevice/createBuffer(descriptor)}} method.
+
+### {{GPUBufferDescriptor}} ### {#GPUBufferDescriptor}
+
+This specifies the options to use in creating a {{GPUBuffer}}.
+The {{GPUBufferDescriptor/size}} and {{GPUBufferDescriptor/usage}} members are required.
+A {{NotFoundError}} exception MUST be thrown if any of the required members are not specified.
 
 <script type=idl>
 dictionary GPUBufferDescriptor : GPUObjectDescriptorBase {
-    required u64 size;
+    required unsigned long long size;
     required GPUBufferUsageFlags usage;
 };
 </script>
 
+<dl dfn-type="abstract-op">
+    : <dfn>validating GPUBufferDescriptor</dfn>(device, descriptor)
+    ::
+        <div algorithm="validation GPUBufferDescriptor(device, descriptor)">
```

Comment:
Some other specs seem to use present progressive tense, "validating GPUBufferDescriptor". Also I don't think the args are needed inside the algorithm name.

Reply:
Thinking about it more, it would make sense to make this be part of the "creating a GPUBuffer" algorithm. May or may not be useful to keep it separate from that.